src/lib/validator.js

```diff
@@ -1,11 +1,34 @@
 const isValidNino = (value) => {
-  return value
-    .replaceAll(" ", "")
-    .toUpperCase()
-    .match(
-      "" ||
-        /^(?!BG|GB|KN|NK|NT|TN|ZZ)[ABCEGHJKLMNOPRSTWXYZ][ABCEGHJKLMNPRSTWXYZ][0-9]{6}[ABCD]$/i
-    );
+  const formattedValue = value.replaceAll(" ", "").toUpperCase();
+  if (formattedValue.length < 9 || formattedValue.length > 9) {
+    return null;
```

Comment:
why not `return false` this will as go with the name `isValidNino` the validators are effectively some kind of test. It better to be clearer. Although JavaScript will treat the presence of value as truthy and the absence of a value or null as `falsey`